app/src/main/java/simform/gitexcercise/android/LoginActivity.kt

```diff
@@ -24,7 +25,9 @@ class LoginActivity : AppCompatActivity() {
             startActivity(Intent(this@LoginActivity, ForgotPasswordActivity::class.java))
         }
         buttonLogin.setOnClickListener {
-            if (etUserName.text.toString().isNotEmpty() && etUserName.text.toString().isNotBlank()) {
+            if (Validator.isValidUserName(etUserName.text.toString()) &&
+                Validator.isValidPassword(etUserPassword.text.toString())
+            ) {
```

Comment:
What about the else part of this condition?
What if only the password is not valid?

Reply:
yes we will have to handle those, made changes. thank you for pointing it out.